tests/api2/test_port_delegates.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python3
+
+import os
+import random
+import sys
+
+apifolder = os.getcwd()
+sys.path.append(apifolder)
+from middlewared.test.integration.utils import call, ValidationErrors
+
+
+PAYLOAD = (
+    ('s3.config', 's3.update', ['bindport', 'console_bindport'], {'access_key': '12345678', 'secret_key': '123456789'}),
+    ('ftp.config', 'ftp.update', ['port'], {}),
+    ('webdav.config', 'webdav.update', ['tcpport', 'tcpportssl'], {}),
+    ('rsyncd.config', 'rsyncd.update', ['port'], {}),
+)
+
+
+def test_port_delegate_validation_with_invalid_ports():
+    in_use_ports = []
+    for entry in call('port.get_in_use'):
+        in_use_ports.extend(filter(lambda i: i > 1024, entry['ports']))
+
+    for config_method, method, keys, payload in PAYLOAD:
+        validation_error = None
+        for key in keys:
+            payload[key] = in_use_ports[random.randint(0, len(in_use_ports) - 1)]
```

Comment:
Why do you use `random`? It's a recipe to non-repeatable failures or false negatives

Reply:
We should be choosing any port which is already being consumed so i wanted to randomly select any element in the list. Should we remove it instead ? ( btw wrt this usage i think we should be safe )